Source/JavaScriptCore/wasm/WasmTypeDefinition.cpp

```diff
@@ -1151,23 +1157,22 @@ void TypeInformation::tryCleanup()
     TypeInformation& info = singleton();
     Locker locker { info.m_lock };
 
-    info.m_unrollingCache.removeIf([&] (auto& keyValuePair) {
-        const TypeDefinition& type = TypeInformation::get(keyValuePair.key);
-        return type.refCount() == 1;
-    });
-
-    info.m_rttMap.removeIf([&] (auto& keyValuePair) {
-        const TypeDefinition& type = TypeInformation::get(keyValuePair.key);
-        return type.refCount() == 1;
-    });
-
-    info.m_typeSet.removeIf([&] (auto& hash) {
-        const auto& signature = hash.key;
-        const bool shouldRemove = signature->refCount() == 1;
-        if (shouldRemove)
-            signature->cleanup();
-        return shouldRemove;
-    });
+    bool changed;
+    do {
+        changed = false;
+        info.m_typeSet.removeIf([&] (auto& hash) {
+            const auto& signature = hash.key;
+            if (signature->refCount() == 1) {
+                TypeIndex index = signature->index();
+                info.m_unrollingCache.remove(index);
+                info.m_rttMap.remove(index);
+                signature->cleanup();
+                changed = true;
```

Comment:
Oops, this was an older version. Should be:

```suggestion
                changed = signature->cleanup();
```